### PR TITLE
Check instruction IsFusible() in gpu_fusible.cc

### PR DESCRIPTION
**Pull request by sana-damani**

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/gpu_fusible.cc [hunk 1]
```diff
@@ -133,17 +133,18 @@ bool ShapesCompatibleForMultiOutputFusion(const HloInstruction& instr1,
 
 bool IsInputFusibleScatter(const HloInstruction& instr) {
   if (instr.opcode() == HloOpcode::kScatter ||
-     (instr.opcode() == HloOpcode::kFusion &&
-      instr.fusion_kind() == HloInstruction::FusionKind::kInput &&
-      instr.fused_expression_root()->opcode() == HloOpcode::kScatter)) {
+      (instr.opcode() == HloOpcode::kFusion &&
+       instr.fusion_kind() == HloInstruction::FusionKind::kInput &&
```
Comment (shahzadlone) [inline tensorflow/compiler/xla/service/gpu/gpu_fusible.cc]: is this your editor adding spaces?

Comment (sana-damani) [reply tensorflow/compiler/xla/service/gpu/gpu_fusible.cc]: I added the spaces to comply with the coding style as referenced here: https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md#c-coding-style

File: tensorflow/compiler/xla/service/gpu/gpu_fusible.cc [hunk 2]
```diff
@@ -133,17 +133,18 @@ bool ShapesCompatibleForMultiOutputFusion(const HloInstruction& instr1,
...
+       instr.fused_expression_root()->opcode() == HloOpcode::kScatter)) {
     return true;
   }
   return false;
 }
 
 bool IsInputFusible(const HloInstruction& instr) {
   // Input fusion only handles non-elemental reduction and scatter operations.
-  return IsInputFusibleReduction(instr) || IsInputFusibleScatter(instr);
+  return instr.IsFusible() &&
```
Comment (thomasjoerg) [inline tensorflow/compiler/xla/service/gpu/gpu_fusible.cc]: Looks like you made the same changes in PR 26277. That makes sense, no need for a separate PR in my opinion.

**Comment by sana-damani** [comment]
Closing as changes included in PR 26277

**Pull request closed without merging**